pkg/server/server.go

```diff
@@ -460,6 +469,7 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 				// TODO: retry with other backends connecting to this agent.
 				klog.ErrorS(err, "CLOSE_REQ to Backend failed", "serverID", s.serverID, "connectionID", connID)
 			}
+			closeReqSent = true
```

Comment:
Here we are setting `closeReqSent = true` even if `backend.Send()` above errored

Reply:
Thanks, good comment! Added a `continue` in the error handling. Thank you!